src/common/snippets/src/op/subgraph.cpp

```diff
@@ -382,11 +382,13 @@ void snippets::op::Subgraph::align_element_types(const BlockedShapeVector& outpu
     //  - Insert Convert before operations that doesn't support original element type for execution
     //  - Insert reverse Convert before operations that support original element type
     //    but have inputs that doesn't support it (because before them will be inserted Convert with exec_type - first point)
-    // Then we should use ConstantFolding pass to convert element type of Scalars before inference.
+    //  - Then we should use ConstantFolding pass to convert element type of Scalars before inference.
+    //  - Eliminate redundant Converts which can be inserted in AlignElementType() pass
     ngraph::pass::Manager manager;
     if (config.m_is_needed_to_align_precision) {
         manager.register_pass<snippets::pass::AlignElementType>(execution_element_type);
         manager.register_pass<ngraph::pass::ConstantFolding>();
+        manager.register_pass<ngraph::pass::EliminateConvert>();
```

Comment:
Thanks a lot for such a detailed response! Again, I think it's important that we save these results in a ticket (just copy&paste your answer), and reference this ticket in the code. Hopefully we'll have some time at the end of the release cycle and will be able to close some of such tickets.
@dmitry-gorokhov , do you think this can be handled in the scope of 99803?

Reply:
I believe that after new logic for precision propagation we can avoid these situations. But to not forget about that, I have created the ticket 100041 anyway

Reply:
@eshoguli please have a look at this thread